packages/nextjs/src/performance/client.ts

```diff
@@ -30,24 +107,27 @@ export function nextRouterInstrumentation(
   startTransactionOnLocationChange: boolean = true,
 ): void {
   startTransaction = startTransactionCb;
-  Router.ready(() => {
-    // We can only start the pageload transaction when we have access to the parameterized
-    // route name. Setting the transaction name after the transaction is started could lead
-    // to possible race conditions with the router, so this approach was taken.
-    if (startTransactionOnPageLoad) {
-      const pathIsRoute = Router.route !== null;
-
-      prevTransactionName = pathIsRoute ? stripUrlQueryAndFragment(Router.route) : global.location.pathname;
-      activeTransaction = startTransactionCb({
-        name: prevTransactionName,
-        op: 'pageload',
-        tags: DEFAULT_TAGS,
-        metadata: {
-          source: pathIsRoute ? 'route' : 'url',
-        },
-      });
-    }
 
+  if (startTransactionOnPageLoad) {
+    const { route, traceParentData, baggage, params } = extractNextDataTagInformation();
+
+    prevTransactionName = route || global.document.location.pathname;
+    const source = route ? 'route' : 'url';
+
+    activeTransaction = startTransactionCb({
+      name: prevTransactionName,
+      op: 'pageload',
+      tags: DEFAULT_TAGS,
+      ...(params && { data: params }),
```

Comment:
iirc setting data does nothing for transactions - can we throw it into some other part of the event? Like a context?

Does this have PII data concerns? Perhaps we guard this with `sendDefaultPII`.

Reply:
Finally found a use for that flag! Yup there are potential pii issues. Added the flag but removed tests around data because that's just not worth the effort of mocking the hub/client/options. 0ba0d6e6f850ab5040980d1ea7186e67687e2f1a

Btw, this data shows up in the trace details when you open a transaction:

![Screen Shot 2022-08-12 at 18 02 39](https://user-images.githubusercontent.com/8118419/184396308-e3b1e4e7-749b-44d8-a922-1c077640d117.png)

Reply:
> Btw, this data shows up in the trace details when you open a transaction:

Ah yeah - that means it'll show up on linked errors as well (since we copy over the entire trace context). I guess since we don't index this we can leave it for now.